tiledb/sm/filter/bit_width_reduction_filter.cc

```diff
@@ -116,9 +116,11 @@ Status BitWidthReductionFilter::run_forward(
   auto data_type_size = static_cast<uint8_t>(datatype_size(filter_data_type_));
 
   // If bit width compression can't work, just return the input unmodified.
-  if ((!datatype_is_integer(filter_data_type_) &&
-       filter_data_type_ != Datatype::BLOB) ||
-      data_type_size == 1) {
+  const bool backwards_compat =
```

Comment:
Would it be cleaner to just move the no-op buffer copy to the default case of the switch statement? If I'm not mistaken, that should work for reading and writing with old filter pipelines while the new `accepts_input_datatype` work prevents new ones from being created. It seems like it could work, but I could also definitely be missing something.

The only other approach I can think of would be to have `accepts_data_type` take a version argument, but that'd obviously require a lot more changes for minimal value I'd think.

Reply:
Nice! This works locally, curious if CI agrees. This check also skips single byte datatypes and that makes sense to me for this filter since we can't reduce the size further. I moved those cases to fall through to default branch, but I still had to check the tile format version for time / datetime branches or some backwards compat tests failed.